specs/flyout/flyout.md

```diff
@@ -0,0 +1,181 @@
+
+### Use Cases
+- As a popup showing additional info for an item on the page.
+- To display a larger image of a thumbnail image in a photo gallery.
+- To display an action related to a page item.
+
+### Features
+**Placement Logic:** - This component will use *Anchored Region* to be positioned relative to the Flyout target element. This way it can be dynamic and adapt to the available space.
+
+**Soft Dismiss (aka Light Dismiss):** - A way to quickly close the Flyout by clicking outside it, or pressing the `esc` key. An event would be emitted anytime the Flyout is dismissed. The Flyout would always be soft-dismissable. Detecting the users next click of a different element can trigger this when clicking outside the Flyout.
+
+**Focus Management:** - Should follow the same process for focus management as a Dialog, primarily as the Flyout's role will be 'dialog'.
+
+This includes:
+- Trap focus - optional, defaults to enabled - Focus is contained within the Flyout while it is visible.
+- First element focus - Focusing on the first focusable element when trap focus is enabled, but not overriding an element that is set to be focused by the author.
+- Returning focus to the target - optional, defaults to enabled, after Flyout is closed the focus would return to the target element.
+
+
+### Prior Art/Examples
+
+#### Popover from Salesforce
+- https://www.lightningdesignsystem.com/components/popovers/
+
+![Popover](./images/salesforce-popover.png)
+#### Popover from Evergreen
+- triggered by a button
+- https://evergreen.segment.com/components/popover/
+
+![Popover](./images/evergreen-popover.png)
+
+#### Microsoft Windows Developer Flyout, Text Only
+![Text Only](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-wrapping-text.png)
+
+#### Microsoft Windows Developer Flyout, Actionable
+![Actionable More Info](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-example2.png)
+
+#### Inline Dialog from ATLAssian
+- https://atlassian.design/components/inline-dialog/examples
+
+![Popover](./images/atlassian-popover.png)
+
+#### Callout with actions from Fluent UI
+- https://developer.microsoft.com/en-us/fluentui#/controls/web/callout
+![Callout](./images/fluent-callout.png)
+
+
+---
+
+### API
+*Component Name*
+- `fast-flyout`
+
+*Attributes*
+- `visible` - boolean, whether or not the flyout is visible, defaults to undefined, as this is controlled by the author
+- `trap-focus` - boolean, whether or not to keep focus contained inside the Flyout, defaults to true
+- `target` - string, the html id of the HTMLElement that the Flyout is attached to, (viewport for anchored region defaults to the parent of the anchor) and triggered by
+- `focus-target-on-close` - boolean, whether or not to focus back on to the _target element_ when the Flyout is closed, defaults to true
```

Comment:
This is definitely open to other naming ideas or approaches to solving this problem, which is described in the focus management section comments

Reply:
After re-reading the spec, it is expected that the target that triggers a dialog is focused on once the dialog is closed and since we have set the role of flyout to dialog we should follow that. It is currently defaulted to do so, but should we allow something like `focus-taget-on-close` or not even make that optional?

Reply:
Maybe intercepting the `dismiss` event to redirect focus somewhere else (or prevent the default focusing behavior) would be the best bet, instead of having an attribute for the behavior?

Reply:
Yeah, I'm not sure that this is valuable here - are there other places where we handle this? Lots of other controls require focus to be placed back onto the invoking element, but I can't think of a place where we are currently handling that with the primitives themselves. Unless we have a more universal story or principle around handling this, I would probably drop this.

Reply:
Yeah, even though it is disruptive to refocus after scrolling, this is still a dialog so it should follow that design pattern.